### PR TITLE
DRAFT :: Enhancement: solve issue #29255 - Change docker image from CentOS to Ubuntu20.04 

### LINKED ISSUE #29255: Change the base docker image to ubuntu 20.04 for Beats & Elastic Agent
**Describe the enhancement:**

Beats and Agent to be aligned with other Elastic products and change the default image to ubuntu 20.04

**Describe a specific use case for the enhancement or feature:**

### PR DESCRIPTION
**Pull request by dsmello**
## What does this PR do?
Chage the docker image from CentOS to Ubuntu20.04 and update the RUN commands.

## Why is it important?
Closes https://github.com/elastic/beats/issues/29255

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Run tests at pipeline

## How to test this PR locally
I don't know, will be tested at pipeline due the scale

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #29255

### CODE REVIEW

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 1]
```diff
@@ -42,27 +42,48 @@ ENV BEAT_SETUID_AS={{ .user }}
 {{- if contains .from "ubi-minimal" }}
 RUN for iter in {1..10}; do microdnf update -y && microdnf install -y shadow-utils jq &&  microdnf clean all && exit_code=0 && break || exit_code=$? && echo "microdnf error: retry $iter in 10s" && sleep 10; done; (exit $exit_code)
 {{- else }}
-# Installing jq needs to be installed after epel-release and cannot be in the same yum install command.
-RUN case $(arch) in aarch64) YUM_FLAGS="-x bind-license";; esac; \
```
Comment (dsmello) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:46]: The YUM_FLAGS was not requested when running ubuntu

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 2]
```diff
@@ -42,27 +42,48 @@ ENV BEAT_SETUID_AS={{ .user }}
...
+RUN for iter in {1..10}; do \
+        apt update -y && \
+        apt install --no-install-recommends --yes jq && \
+        apt clean all && \
+        exit_code=0 && break || exit_code=$? && echo "apt error: retry $iter in 10s" && sleep 10; \
     done; \
     (exit $exit_code)
 {{- end }}
 
+## Changes Notes: To migratre from CentOS to Ubuntu, the following changes are needed:
```
Comment (dsmello) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:55]: Those are all packages migrated from CentOs to Ubuntu related

**Comment by dsmello** [comment]
run elasticsearch-ci/docs

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl (hunk 2 above)
Comment (ruflin) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:55]: Is this temporarily for discussion or you plane to level this in here?

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 3]
```diff
@@ -42,27 +42,48 @@ ENV BEAT_SETUID_AS={{ .user }}
...
+# libXext -> libxext-dev 
+# libXi -> libxi-dev
+# libXtst -> libxtst-dev
+# libXScrnSaver -> libxss-dev
+# libXrandr -> libxrandr-dev
+# GConf2 -> libgconf2-4
+# alsa-lib -> libasound2
+# gtk3 -> GTK3 is installed by default REMOVED
+# ipa-gothic-fonts -> fonts-ipafont-gothic
+# xorg-x11-fonts-100dpi -> xfonts-100dpi
```
Comment (ruflin) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:73]: @jsoriano @andrewkroh Do you remember what requires the xorg dependencies?

Comment (mtojek) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:73]: Blind guess: abrt-xorg.

> The abrt-xorg service collects and processes information about crashes of the X.Org server.

Comment (andrewkroh) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:73]: That must be a dependency required by Synthetics.

Comment (ruflin) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:73]: Ah, I missed the part below that it is only included in the "complete" image. Good to know it is not in the standard image.

@andrewvc FYI

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl (hunk 2 above)
Comment (dsmello) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:55]: Hello @ruflin, this is for temporary discussion, I'll try to find where I add them to a changelog for future notes. 😸

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl (hunk 3 above)
Comment (andrewvc) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:73]: Yep!

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 4]
```diff
@@ -42,27 +42,48 @@ ENV BEAT_SETUID_AS={{ .user }}
...
-        yum update -y $YUM_FLAGS && \
-        yum install -y epel-release &&  \
-        yum update -y $YUM_FLAGS && \
-        yum install -y jq && \
-        yum clean all && \
-        exit_code=0 && break || exit_code=$? && echo "yum error: retry $iter in 10s" && sleep 10; \
+
+RUN for iter in {1..10}; do \
+        apt update -y && \
+        apt install --no-install-recommends --yes jq && \
```
Comment (blakerouse) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:48]: You need to add `DEBIAN_FRONTEND=noninteractive` in front of the `apt install` command here.

So it should be come:

`DEBIAN_FRONTEND=noninteractive apt install --no-install-recommends --yes jq`

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 5]
```diff
@@ -42,27 +42,48 @@ ENV BEAT_SETUID_AS={{ .user }}
...
-RUN for iter in {1..10}; do \
-        yum -y install atk gtk gdk xrandr pango libXcomposite libXcursor libXdamage \
-        libXext libXi libXtst libXScrnSaver libXrandr GConf2 \
-        alsa-lib atk gtk3 ipa-gothic-fonts xorg-x11-fonts-100dpi xorg-x11-fonts-75dpi xorg-x11-utils \
-        xorg-x11-fonts-cyrillic xorg-x11-fonts-Type1 xorg-x11-fonts-misc \
-        yum clean all && \
-        exit_code=0 && break || exit_code=$? && echo "yum error: retry $iter in 10s" && sleep 10; \
+RUN apt update -y && \
+    for iter in {1..10}; do \
+        apt install --no-install-recommends --yes libatk1.0-0 libgdk-pixbuf2.0-dev x11-xserver-utils libpango-1.0-0 libxcomposite-dev libxcursor-dev libxdamage-dev \
```
Comment (blakerouse) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:82]: Same here add the `DEBIAN_FRONTEND=noninteractive`

**Comment by blakerouse** [comment]
Closing in favor of #29681

**Pull request closed without merging**